From a726db8e1a66d6374eead53dbbb30a0f779cc410 Mon Sep 17 00:00:00 2001
From: Samuel Mimram <smimram@gmail.com>
Date: Wed, 25 May 2016 10:31:33 +0200
Subject: [PATCH 2/6] Remove duplicate declaration of write.

---
 src/ssl.mli | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/src/ssl.mli b/src/ssl.mli
index 4f67dd2..5776328 100644
--- a/src/ssl.mli
+++ b/src/ssl.mli
@@ -395,9 +395,6 @@ val file_descr_of_socket : socket -> Unix.file_descr
 (** [read sock buf off len] receives data from a connected SSL socket. *)
 val read : socket -> Bytes.t -> int -> int -> int
 
-(** [write sock buf off len] sends data over a connected SSL socket. *)
-val write : socket -> Bytes.t -> int -> int -> int
-
 (** [read_into_bigarray sock ba off len] receives data from a connected SSL socket.
     This function releases the runtime while the read takes place. *)
 val read_into_bigarray : socket -> bigarray -> int -> int -> int
@@ -409,7 +406,7 @@ val read_into_bigarray : socket -> bigarray -> int -> int -> int
 val read_into_bigarray_blocking : socket -> bigarray -> int -> int -> int
 
 (** [write sock buf off len] sends data over a connected SSL socket. *)
-val write : socket -> string -> int -> int -> int
+val write : socket -> Bytes.t -> int -> int -> int
 
 (** [write_bigarray sock ba off len] sends data over a connected SSL socket.
     This function releases the runtime while the read takes place.
-- 
2.9.3

