From f94658b844464945bfeef4396fc2461943b8b964 Mon Sep 17 00:00:00 2001
From: Samuel Mimram <smimram@gmail.com>
Date: Thu, 28 Jul 2016 10:27:30 +0200
Subject: [PATCH 3/6] Bump version number.

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 853e9a0..9613001 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,4 +1,4 @@
-AC_INIT([ocaml-ssl],[0.5.2],[savonet-users@lists.sourceforge.net])
+AC_INIT([ocaml-ssl],[0.5.3],[savonet-users@lists.sourceforge.net])
 
 VERSION=$PACKAGE_VERSION
 AC_MSG_RESULT([configuring $PACKAGE_STRING])
-- 
2.9.3

