From e49b6a741ac7a86057acb7bdef609e67491b292e Mon Sep 17 00:00:00 2001
From: Samuel Mimram <smimram@gmail.com>
Date: Thu, 28 Jul 2016 12:34:31 +0200
Subject: [PATCH 5/6] Use accessor functions for X509_STORE_CTX.

---
 CHANGES         | 1 +
 src/ssl_stubs.c | 9 +++++----
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/CHANGES b/CHANGES
index 7fa5ec1..87bd915 100644
--- a/CHANGES
+++ b/CHANGES
@@ -1,6 +1,7 @@
 0.5.3 (unreleased)
 =====
 * Remove -ansi flag to be compatible with OCaml 4.04 (thanks Mark Shinwell).
+* Use accessor functions for X509_STORE_CTX.
 
 0.5.2 (2015-11-23)
 =====
diff --git a/src/ssl_stubs.c b/src/ssl_stubs.c
index 1b175fa..e131527 100644
--- a/src/ssl_stubs.c
+++ b/src/ssl_stubs.c
@@ -656,7 +656,7 @@ CAMLprim value ocaml_ssl_get_cipher_name(value vcipher)
 
 CAMLprim value ocaml_ssl_get_cipher_version(value vcipher)
 {
-  char *version;
+  const char *version;
   SSL_CIPHER *cipher = (SSL_CIPHER*)vcipher;
 
   caml_enter_blocking_section();
@@ -1228,9 +1228,9 @@ static int client_verify_callback(int ok, X509_STORE_CTX *ctx)
   int depth, error;
   char *xs;
 
-  depth = ctx->error_depth;
-  error = ctx->error;
-  xs = (char *)X509_STORE_CTX_get_current_cert(ctx);
+  depth = X509_STORE_CTX_get_error_depth(ctx);
+  error = X509_STORE_CTX_get_error(ctx);
+  xs = (char*)X509_STORE_CTX_get_current_cert(ctx);
 
   subject = issuer = NULL;
 
@@ -1339,6 +1339,7 @@ return_time:
 
   return ok;
 }
+
 static DH *load_dh_param(const char *dhfile)
 {
   DH *ret=NULL;
-- 
2.9.3

